bin/replicate-mongodb.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ "$#" == "0" ]]; then
+  echo "usage: $0 \$app"
+  exit 1
+fi
+
+app="$1"
+
+if echo "$app" | grep -q '_'; then
```

Comment:
Could we automatically convert underscores into dashes (while telling the user what's happening)?

This feels like one of those times when the script understands what you were trying to do but refuses to do so anyway, similar to https://github.com/alphagov/fabric-scripts/pull/285.

Reply:
I'm cautious about this kind of thing. Although it's useful, it's also good to encourage people to know the correct names, for when there isn't a script to help them. Perhaps there's a case when it's particularly unintuitive?

Reply:
I'm also generally cautious of these kinds of things, but I think in this case it's reasonable since the behaviour is very predictable. I think it's better than the previous situation where the script understands the error the user has made and how to correct it, but makes no attempt to correct it (which is a frustrating user experience).

Perhaps an alternative is that the script errors saying something like `Could not find database backup for "content_store" to replicate` after trying to download the backup, rather than doing a preemptive check. This has the benefit of showing the same error if someone misspells the app name or if they use an underscore.

Reply:
I agree - it's very similar to rubocop (not) autocorrecting stuff, which used to infuriate me. My issue with rubocop went away when I realised it was just telling me what I should be typing in the first place. But I understand that's not how everyone sees it. I like your suggestion, as that error handler sounds more useful in other cases (mis-spellings).

Reply:
The `aws s3 cp` commands will already fail saying something about being unable to find the object, is a more specific error message needed?